azure-kusto-data/source/managedIdentitiesClient.js

```diff
@@ -1,39 +1,25 @@
-const http = require("http");
-const url = require("url");
-
-module.exports = function acquireToken(resource, msiEndpoint, msiSecret, callback) {
-    const msiResource = `${msiEndpoint}/?resource=${resource}&api-version=2017-09-01`;
-    const msiUrl = url.parse(msiResource);
-
-    const options = {
-        host: msiUrl.host,
-        hostname: msiUrl.hostname,
-        port: msiUrl.port,
-        path: msiUrl.path,
-        headers: {
-            secret: msiSecret
+const request = require("request");
+
+module.exports = function acquireToken(resource, msiEndpoint, msiClientId, callback) {
```

Comment:
I don't know if this is going to fly for Azure Functions and App Services as stated on this [documentation](https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity), it requires to use the `secret` header for the request.

Reply:
Thanks for pointing it out! I missed the fact that there is a separate and somewhat different use when it comes to Azure Functions.

If that is the case, wouldn't it make more sense to check for those env variables and do the resolving instead of specific assignment by the end user?

Reply:
I guess it would make sense. I decided before to add the feature as assignment as I don't know the whole implementation details of manage identities. I wasn't sure if there's a way to mock the url for example when running it local.

Based on that logic, I prefer the user to set it up to allow some flexibility. But for my scenario I wouldn't mind since I have not been able to test it locally anyway.